OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -499,12 +499,11 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 
 					// Try and place the refinery near a resource field
 					var nearbyResources = Map.FindTilesInCircle(baseCenter, Info.MaxBaseRadius)
-						.Where(a => resourceTypeIndices.Get(Map.GetTerrainIndex(a)))
-						.Shuffle(Random);
+						.Where(a => resourceTypeIndices.Get(Map.GetTerrainIndex(a)));
 
-					foreach (var c in nearbyResources)
+					if (nearbyResources.Any())
 					{
-						var found = findPos(c, baseCenter, 0, Info.MaxBaseRadius);
+						var found = findPos(nearbyResources.RandomOrDefault(Random), baseCenter, 0, Info.MaxBaseRadius);
```

Comment:
We're back to the double enumeration problem. You should consider a ```ToList``` or my earlier suggestion in IRC to cast to a ```CPos?``` so RandomOrDefault works better for you.